sympy/matrices/matrices.py

```diff
@@ -3981,3 +3981,69 @@ def classof(A, B):
 def a2idx(j, n=None):
     from sympy.matrices.common import a2idx as a2idx_
     return a2idx_(j, n)
+
+def upper(a, diag=0):
+    from sympy import Matrix
+    """
+    Examples
+    ========
+
+    >>> from sympy.matrices.matrices import upper
```

Comment:
I think that user examples should better use the imports from the global namespace.
And also to make sure that you have added this to the `sympy.matrices.__init__`

Reply:
where to add the function in `__init__` should i write it in `__all__ = [.....]`?

Reply:
> I think that user examples should better use the imports from the global namespace.

can you explain it please?

Reply:
> can you explain it please?

I mean that since it was often annoying that some functions are not importable with `from sympy import foo`.
I think that every user API should be importable under the global namespace. I'm even thinking of banning the imports that are too specific, for user APIs or documentations.

You should add things to this function to `matrices/__init__.py`. But regarding the comments from @asmeurer you may not even have to do that if you just add this under the `MatrixBase` class.

Reply:
> I mean that since it was often annoying that some functions are not importable with `from sympy import foo`.
> I think that every user API should be importable under the global namespace. I'm even thinking of banning the imports that are too specific, for user APIs or documentations.

Yes I agree with you

Reply:
ok I have done the required changes. Can you have a look? @sylee957